lib/std/fs/file.zig

```diff
@@ -385,6 +385,274 @@ pub const File = struct {
+            /// This method *DOES NOT* set permissions on the filesystem: use `File.setPermissions(permissions)`
+            pub fn setReadOnly(self: *Self, read_only: bool) void {
+                if (read_only) {
+                    self.mode &= ~@as(Mode, 0o222);
+                } else {
+                    self.mode |= @as(Mode, 0o222);
+                }
+            }
+
+            pub const Class = enum(u2) {
+                user = 2,
+                group = 1,
+                other = 0,
+            };
+
+            pub const Permission = enum(u3) {
+                read = 0o4,
+                write = 0o2,
+                execute = 0o1,
+            };
+
+            /// Returns `true` if the chosen class has the selected permission.
+            /// This method is only available on Unix platforms.
+            pub fn unixHas(self: *Self, class: Class, permission: Permission) bool {
+                const mask = @as(Mode, @enumToInt(permission)) << @as(u3, @enumToInt(class)) * 3;
+                return self.mode & mask != 0;
+            }
+
+            /// Sets the permissions for the chosen class. Any permissions set to `null` are left unchanged.
+            /// This method *DOES NOT* set permissions on the filesystem: use `File.setPermissions(permissions)`
+            /// This method is only available on Unix platforms.
+            pub fn unixSet(self: *Self, class: Class, permissions: struct {
+                read: ?bool = null,
+                write: ?bool = null,
+                execute: ?bool = null,
+            }) void {
+                const shift = @as(u3, @enumToInt(class)) * 3;
+                if (permissions.read) |r| {
+                    if (r) {
+                        self.mode |= @as(Mode, 0o4) << shift;
+                    } else {
+                        self.mode &= ~(@as(Mode, 0o4) << shift);
+                    }
+                }
+                if (permissions.write) |w| {
+                    if (w) {
+                        self.mode |= @as(Mode, 0o2) << shift;
+                    } else {
+                        self.mode &= ~(@as(Mode, 0o2) << shift);
+                    }
+                }
+                if (permissions.execute) |x| {
+                    if (x) {
+                        self.mode |= @as(Mode, 0o1) << shift;
+                    } else {
+                        self.mode &= ~(@as(Mode, 0o1) << shift);
+                    }
+                }
+            }
+
+            /// Returns a `Permissions` struct representing the permissions from the passed mode.
+            /// This method is only available on Unix platforms.
+            pub fn unixNew(new_mode: Mode) Self {
+                return Self{
+                    .mode = new_mode,
+                };
+            }
+        },
+    };
+
+    pub const SetPermissionsError = if (is_windows) error{AccessDenied} || os.UnexpectedError else ChmodError;
```

Comment:
```suggestion
    pub const SetPermissionsError = ChmodError;
```
The error set should contain all possible errors for all possible targets.

Reply:
Not sure why you're suggesting to remove the possible errors for Windows...

Reply:
Oh, I see now. `FChmodError`, by coincidence, contains the possible errors thrown on Windows. Is this a sensible thing to assume though?

Reply:
Actually that makes sense, I'm just way too tired to be working on this